README.md
# Go API client for UNPACME


# Introduction
Welcome to the UNPACME API! All the malware unpacking and file analysis features that you are familiar with on the [unpac.me](https://www.unpac.me/) website are available through our API. You can easily integrate our unpacker into your malware analysis pipeline and begin unpacking at scale!


# Authentication
The public UNPACME API is publicly available and can be accessed without authentication.

In order to use the private UNPACME API you must sign up for an account with UNPACME. Once you have a valid user account you can view your personal API key in your user profile. 

<SecurityDefinitions />

# Response Structure
When interacting with the UNPACME API, if the request was correctly handled, a <b>200</b> HTTP status code will be returned. The body of the response will usually be a JSON object (except for file downloads).

## Response Status Codes

Status Code  | Description | Notes
------------- | ------------- | -
200  | OK | The request was successful
400  | Bad Request | The request was somehow incorrect. This can be caused by missing arguments or arguments with wrong values.
401 | Unauthorized | The supplied credentials, if any, are not sufficient to access the resource
403 | Forbidden | The account does not have enough privileges to make the request.
404 | Not Found | The requested resource is not found
429  | Too Many Requests | The request frequency has exceeded one of the account quotas (minute, daily or monthly). Monthly quotas are reset on the 1st of the month at 00:00 UTC.
500 | Server Error | The server could not return the representation due to an internal server error


## Error Response

If an error has occurred while handling the request an error status code will be returend along with a JSON error message with the following properties.


Property  | Description
------------- | -------------
Error  | The error type
Description  | A more informative message


## Overview
This API client was generated by the [OpenAPI Generator](https://openapi-generator.tech) project.  By using the [OpenAPI-spec](https://www.openapis.org/) from a remote server, you can easily generate an API client.

- API version: 1.0.0
- Package version: 1.0.0
- Build package: org.openapitools.codegen.languages.GoClientCodegen

## Installation

Install the following dependencies:

```shell
go get github.com/stretchr/testify/assert
go get golang.org/x/oauth2
go get golang.org/x/net/context
```

Install the module:
```
go get github.com/kryptoslogic/unpacme-go
```

Import:

```golang
import openapiclient "github.com/kryptoslogic/unpacme-go"
```

To use a proxy, set the environment variable `HTTP_PROXY`:

```golang
os.Setenv("HTTP_PROXY", "http://proxy_name:proxy_port")
```

## Documentation for API Endpoints

All URIs are relative to *https://api.unpac.me/api/v1*

Class | Method | HTTP request | Description
------------ | ------------- | ------------- | -------------
*FeedApi* | [**GetPrivateFeed**](docs/FeedApi.md#getprivatefeed) | **Get** /private/feed/unpacked | Get full unpacked sample feed
*FeedApi* | [**GetPrivateFeedYaraFiltered**](docs/FeedApi.md#getprivatefeedyarafiltered) | **Get** /private/feed/unpacked/yara/{yara_rule} | Get full unpacked sample feed filtered by yara rule
*FeedApi* | [**GetPrivateFeedYaraTags**](docs/FeedApi.md#getprivatefeedyaratags) | **Get** /private/feed/unpacked/yara | Get list of yara tags in feed
*PublicApi* | [**GetPublicFeed**](docs/PublicApi.md#getpublicfeed) | **Get** /public/feed | Get public feed
*PublicApi* | [**GetPublicResults**](docs/PublicApi.md#getpublicresults) | **Get** /public/results/{unpack_id} | Get unpack results by ID
*PublicApi* | [**GetPublicUnpackStatus**](docs/PublicApi.md#getpublicunpackstatus) | **Get** /public/status/{unpack_id} | Get unpack status by ID
*UnpackingApi* | [**GetPrivateDownload**](docs/UnpackingApi.md#getprivatedownload) | **Get** /private/download/{sample_hash} | Download sample by hash
*UnpackingApi* | [**GetPrivateHistory**](docs/UnpackingApi.md#getprivatehistory) | **Get** /private/history | Get history
*UnpackingApi* | [**GetPrivateResults**](docs/UnpackingApi.md#getprivateresults) | **Get** /private/results/{unpack_id} | Get unpack results by ID
*UnpackingApi* | [**GetPrivateSearchbyHash**](docs/UnpackingApi.md#getprivatesearchbyhash) | **Get** /private/search/hash/{sample_hash} | Search for parent submission by hash
*UnpackingApi* | [**GetPrivateUnpackStatus**](docs/UnpackingApi.md#getprivateunpackstatus) | **Get** /private/status/{unpack_id} | Get unpack status by ID
*UnpackingApi* | [**PostPrivateUpload**](docs/UnpackingApi.md#postprivateupload) | **Post** /private/upload/ | Submit sample for unpacking
*UserApi* | [**DeletePrivateUserMalpedia**](docs/UserApi.md#deleteprivateusermalpedia) | **Delete** /private/user/malpedia | Remove Malpedia authentication
*UserApi* | [**GetPrivateUserAccess**](docs/UserApi.md#getprivateuseraccess) | **Get** /private/user/access | Get user settings
*UserApi* | [**GetPrivateUserMalpedia**](docs/UserApi.md#getprivateusermalpedia) | **Get** /private/user/malpedia | Get user Malpedia info
*UserApi* | [**PostPrivateUserMalpedia**](docs/UserApi.md#postprivateusermalpedia) | **Post** /private/user/malpedia | Authenticate user to Malpedia


## Documentation For Models

 - [DeepmatchEntity](docs/DeepmatchEntity.md)
 - [DeepmatchEntityAllOf](docs/DeepmatchEntityAllOf.md)
 - [DetectitEntity](docs/DetectitEntity.md)
 - [DetectitEntityAllOf](docs/DetectitEntityAllOf.md)
 - [Export](docs/Export.md)
 - [ExportAllOf](docs/ExportAllOf.md)
 - [FeedEntity](docs/FeedEntity.md)
 - [Function](docs/Function.md)
 - [FunctionAllOf](docs/FunctionAllOf.md)
 - [History](docs/History.md)
 - [HistoryEntity](docs/HistoryEntity.md)
 - [ImportEntity](docs/ImportEntity.md)
 - [ImportEntityAllOf](docs/ImportEntityAllOf.md)
 - [InlineObject](docs/InlineObject.md)
 - [InlineObject1](docs/InlineObject1.md)
 - [InlineResponse200](docs/InlineResponse200.md)
 - [InlineResponse2001](docs/InlineResponse2001.md)
 - [MalwareId](docs/MalwareId.md)
 - [MalwareIdAllOf](docs/MalwareIdAllOf.md)
 - [MalwareIdShort](docs/MalwareIdShort.md)
 - [MalwareIdShortAllOf](docs/MalwareIdShortAllOf.md)
 - [PrivateFeed](docs/PrivateFeed.md)
 - [PrivateFeedEntity](docs/PrivateFeedEntity.md)
 - [PrivateFeedEntityChildren](docs/PrivateFeedEntityChildren.md)
 - [PrivateFeedFiltered](docs/PrivateFeedFiltered.md)
 - [PrivateFeedYaraTags](docs/PrivateFeedYaraTags.md)
 - [PublicFeed](docs/PublicFeed.md)
 - [Resource](docs/Resource.md)
 - [ResourceAllOf](docs/ResourceAllOf.md)
 - [ResourceEntity](docs/ResourceEntity.md)
 - [ResourceEntry](docs/ResourceEntry.md)
 - [Result](docs/Result.md)
 - [ResultAllOf](docs/ResultAllOf.md)
 - [ResultAllOfAnalysis](docs/ResultAllOfAnalysis.md)
 - [ResultAllOfAnalysisExports](docs/ResultAllOfAnalysisExports.md)
 - [ResultAllOfAnalysisImports](docs/ResultAllOfAnalysisImports.md)
 - [ResultAllOfAnalysisMetadata](docs/ResultAllOfAnalysisMetadata.md)
 - [ResultAllOfAnalysisMetadataVersionInfo](docs/ResultAllOfAnalysisMetadataVersionInfo.md)
 - [ResultAllOfAnalysisMetadataVersionInfoStringInfo](docs/ResultAllOfAnalysisMetadataVersionInfoStringInfo.md)
 - [ResultAllOfAnalysisMetadataVersionInfoVarInfo](docs/ResultAllOfAnalysisMetadataVersionInfoVarInfo.md)
 - [ResultAllOfAnalysisRichHeaders](docs/ResultAllOfAnalysisRichHeaders.md)
 - [ResultAllOfHashes](docs/ResultAllOfHashes.md)
 - [ResultAllOfStrings](docs/ResultAllOfStrings.md)
 - [RichHeader](docs/RichHeader.md)
 - [RichHeaderAllOf](docs/RichHeaderAllOf.md)
 - [SearchEntity](docs/SearchEntity.md)
 - [SearchResults](docs/SearchResults.md)
 - [Section](docs/Section.md)
 - [SectionAllOf](docs/SectionAllOf.md)
 - [Status](docs/Status.md)
 - [UnpackResults](docs/UnpackResults.md)
 - [UnpackResultsAllOf](docs/UnpackResultsAllOf.md)
 - [UnpackStatus](docs/UnpackStatus.md)
 - [UnpackStatusAllOf](docs/UnpackStatusAllOf.md)
 - [UserAccess](docs/UserAccess.md)
 - [UserAccessAllOf](docs/UserAccessAllOf.md)


## Documentation For Authorization



### api_key

- **Type**: API key
- **API key parameter name**: Authorization
- **Location**: HTTP header

Note, each API key must be added to a map of `map[string]APIKey` where the key is: Authorization and passed in as the auth context for each request.


## Documentation for Utility Methods

Due to the fact that model structure members are all pointers, this package contains
a number of utility functions to easily obtain pointers to values of basic types.
Each of these functions takes a value of the given basic type and returns a pointer to it:

* `PtrBool`
* `PtrInt`
* `PtrInt32`
* `PtrInt64`
* `PtrFloat`
* `PtrFloat32`
* `PtrFloat64`
* `PtrString`
* `PtrTime`

## Author


Kryptos Logic


docs/History.md
# History

## Properties

Name | Type | Description | Notes
------------ | ------------- | ------------- | -------------
**Cursor** | Pointer to **int32** | Cursor used to scroll to page of results | [optional] 
**Results** | Pointer to [**[]HistoryEntity**](HistoryEntity.md) |  | [optional] 

## Methods

### NewHistory

`func NewHistory() *History`

NewHistory instantiates a new History object
This constructor will assign default values to properties that have it defined,
and makes sure properties required by API are set, but the set of arguments
will change when the set of required properties is changed

### NewHistoryWithDefaults

`func NewHistoryWithDefaults() *History`

NewHistoryWithDefaults instantiates a new History object
This constructor will only assign default values to properties that have it defined,
but it doesn't guarantee that properties required by API are set

### GetCursor

`func (o *History) GetCursor() int32`

GetCursor returns the Cursor field if non-nil, zero value otherwise.

### GetCursorOk

`func (o *History) GetCursorOk() (*int32, bool)`

GetCursorOk returns a tuple with the Cursor field if it's non-nil, zero value otherwise
and a boolean to check if the value has been set.

### SetCursor

`func (o *History) SetCursor(v int32)`

SetCursor sets Cursor field to given value.

### HasCursor

`func (o *History) HasCursor() bool`

HasCursor returns a boolean if a field has been set.

### GetResults

`func (o *History) GetResults() []HistoryEntity`

GetResults returns the Results field if non-nil, zero value otherwise.

### GetResultsOk

`func (o *History) GetResultsOk() (*[]HistoryEntity, bool)`

GetResultsOk returns a tuple with the Results field if it's non-nil, zero value otherwise
and a boolean to check if the value has been set.

### SetResults

`func (o *History) SetResults(v []HistoryEntity)`

SetResults sets Results field to given value.

### HasResults

`func (o *History) HasResults() bool`

HasResults returns a boolean if a field has been set.


[[Back to Model list]](../README.md#documentation-for-models) [[Back to API list]](../README.md#documentation-for-api-endpoints) [[Back to README]](../README.md)




api_feed.go
/*
 * UnpacMe
 *
 *  # Introduction Welcome to the UNPACME API! All the malware unpacking and file analysis features that you are familiar with on the [unpac.me](https://www.unpac.me/) website are available through our API. You can easily integrate our unpacker into your malware analysis pipeline and begin unpacking at scale!   # Authentication The public UNPACME API is publicly available and can be accessed without authentication.  In order to use the private UNPACME API you must sign up for an account with UNPACME. Once you have a valid user account you can view your personal API key in your user profile.   <SecurityDefinitions />  # Response Structure When interacting with the UNPACME API, if the request was correctly handled, a <b>200</b> HTTP status code will be returned. The body of the response will usually be a JSON object (except for file downloads).  ## Response Status Codes  Status Code  | Description | Notes ------------- | ------------- | - 200  | OK | The request was successful 400  | Bad Request | The request was somehow incorrect. This can be caused by missing arguments or arguments with wrong values. 401 | Unauthorized | The supplied credentials, if any, are not sufficient to access the resource 403 | Forbidden | The account does not have enough privileges to make the request. 404 | Not Found | The requested resource is not found 429  | Too Many Requests | The request frequency has exceeded one of the account quotas (minute, daily or monthly). Monthly quotas are reset on the 1st of the month at 00:00 UTC. 500 | Server Error | The server could not return the representation due to an internal server error   ## Error Response  If an error has occurred while handling the request an error status code will be returend along with a JSON error message with the following properties.   Property  | Description ------------- | ------------- Error  | The error type Description  | A more informative message  # Example Clients  The following clients can be used to interact with the UNPACME API directly and are provided as examples. These clients are community projects and are not maintained or developed by UNPACME. UNPACME makes no claim as to the safety of these clients, use at your own risk.    - [UnpacMe Client](https://github.com/larsborn/UnpacMeClient) (Python)   - [UnpacMe Library](https://github.com/R3MRUM/unpacme) (Python)  <br> 
 *
 * API version: 1.0.0
 */

// Code generated by OpenAPI Generator (https://openapi-generator.tech); DO NOT EDIT.

package openapi

import (
	_context "context"
	_ioutil "io/ioutil"
	_nethttp "net/http"
	_neturl "net/url"
	"strings"
)

// Linger please
var (
	_ _context.Context
)

// FeedApiService FeedApi service
type FeedApiService service

type ApiGetPrivateFeedRequest struct {
	ctx _context.Context
	ApiService *FeedApiService
	cursor *int32
}

func (r ApiGetPrivateFeedRequest) Cursor(cursor int32) ApiGetPrivateFeedRequest {
	r.cursor = &cursor
	return r
}

func (r ApiGetPrivateFeedRequest) Execute() (PrivateFeed, *_nethttp.Response, error) {
	return r.ApiService.GetPrivateFeedExecute(r)
}

/*
 * GetPrivateFeed Get full unpacked sample feed
 * Returns full feed of unpacked samples
 * @param ctx _context.Context - for authentication, logging, cancellation, deadlines, tracing, etc. Passed from http.Request or context.Background().
 * @return ApiGetPrivateFeedRequest
 */
func (a *FeedApiService) GetPrivateFeed(ctx _context.Context) ApiGetPrivateFeedRequest {
	return ApiGetPrivateFeedRequest{
		ApiService: a,
		ctx: ctx,
	}
}

/*
 * Execute executes the request
 * @return PrivateFeed
 */
func (a *FeedApiService) GetPrivateFeedExecute(r ApiGetPrivateFeedRequest) (PrivateFeed, *_nethttp.Response, error) {
	var (
		localVarHTTPMethod   = _nethttp.MethodGet
		localVarPostBody     interface{}
		localVarFormFileName string
		localVarFileName     string
		localVarFileBytes    []byte
		localVarReturnValue  PrivateFeed
	)

	localBasePath, err := a.client.cfg.ServerURLWithContext(r.ctx, "FeedApiService.GetPrivateFeed")
	if err != nil {
		return localVarReturnValue, nil, GenericOpenAPIError{error: err.Error()}
	}

	localVarPath := localBasePath + "/private/feed/unpacked"

	localVarHeaderParams := make(map[string]string)
	localVarQueryParams := _neturl.Values{}
	localVarFormParams := _neturl.Values{}

	if r.cursor != nil {
		localVarQueryParams.Add("cursor", parameterToString(*r.cursor, ""))
	}
	// to determine the Content-Type header
	localVarHTTPContentTypes := []string{}

	// set Content-Type header
	localVarHTTPContentType := selectHeaderContentType(localVarHTTPContentTypes)
	if localVarHTTPContentType != "" {
		localVarHeaderParams["Content-Type"] = localVarHTTPContentType
	}

	// to determine the Accept header
	localVarHTTPHeaderAccepts := []string{"application/json"}

	// set Accept header
	localVarHTTPHeaderAccept := selectHeaderAccept(localVarHTTPHeaderAccepts)
	if localVarHTTPHeaderAccept != "" {
		localVarHeaderParams["Accept"] = localVarHTTPHeaderAccept
	}
	if r.ctx != nil {
		// API Key Authentication
		if auth, ok := r.ctx.Value(ContextAPIKeys).(map[string]APIKey); ok {
			if apiKey, ok := auth["api_key"]; ok {
				var key string
				if apiKey.Prefix != "" {
					key = apiKey.Prefix + " " + apiKey.Key
				} else {
					key = apiKey.Key
				}
				localVarHeaderParams["Authorization"] = key
			}
		}
	}
	req, err := a.client.prepareRequest(r.ctx, localVarPath, localVarHTTPMethod, localVarPostBody, localVarHeaderParams, localVarQueryParams, localVarFormParams, localVarFormFileName, localVarFileName, localVarFileBytes)
	if err != nil {
		return localVarReturnValue, nil, err
	}

	localVarHTTPResponse, err := a.client.callAPI(req)
	if err != nil || localVarHTTPResponse == nil {
		return localVarReturnValue, localVarHTTPResponse, err
	}

	localVarBody, err := _ioutil.ReadAll(localVarHTTPResponse.Body)
	localVarHTTPResponse.Body.Close()
	if err != nil {
		return localVarReturnValue, localVarHTTPResponse, err
	}

	if localVarHTTPResponse.StatusCode >= 300 {
		newErr := GenericOpenAPIError{
			body:  localVarBody,
			error: localVarHTTPResponse.Status,
		}
		return localVarReturnValue, localVarHTTPResponse, newErr
	}

	err = a.client.decode(&localVarReturnValue, localVarBody, localVarHTTPResponse.Header.Get("Content-Type"))
	if err != nil {
		newErr := GenericOpenAPIError{
			body:  localVarBody,
			error: err.Error(),
		}
		return localVarReturnValue, localVarHTTPResponse, newErr
	}

	return localVarReturnValue, localVarHTTPResponse, nil
}

type ApiGetPrivateFeedYaraFilteredRequest struct {
	ctx _context.Context
	ApiService *FeedApiService
	yaraRule string
}


func (r ApiGetPrivateFeedYaraFilteredRequest) Execute() (PrivateFeedFiltered, *_nethttp.Response, error) {
	return r.ApiService.GetPrivateFeedYaraFilteredExecute(r)
}

/*
 * GetPrivateFeedYaraFiltered Get full unpacked sample feed filtered by yara rule
 * Returns full feed of unpacked samples filtered by the yara rule
 * @param ctx _context.Context - for authentication, logging, cancellation, deadlines, tracing, etc. Passed from http.Request or context.Background().
 * @param yaraRule Yara rule name used to filter feed
 * @return ApiGetPrivateFeedYaraFilteredRequest
 */
func (a *FeedApiService) GetPrivateFeedYaraFiltered(ctx _context.Context, yaraRule string) ApiGetPrivateFeedYaraFilteredRequest {
	return ApiGetPrivateFeedYaraFilteredRequest{
		ApiService: a,
		ctx: ctx,
		yaraRule: yaraRule,
	}
}

/*
 * Execute executes the request
 * @return PrivateFeedFiltered
 */
func (a *FeedApiService) GetPrivateFeedYaraFilteredExecute(r ApiGetPrivateFeedYaraFilteredRequest) (PrivateFeedFiltered, *_nethttp.Response, error) {
	var (
		localVarHTTPMethod   = _nethttp.MethodGet
		localVarPostBody     interface{}
		localVarFormFileName string
		localVarFileName     string
		localVarFileBytes    []byte
		localVarReturnValue  PrivateFeedFiltered
	)

	localBasePath, err := a.client.cfg.ServerURLWithContext(r.ctx, "FeedApiService.GetPrivateFeedYaraFiltered")
	if err != nil {
		return localVarReturnValue, nil, GenericOpenAPIError{error: err.Error()}
	}

	localVarPath := localBasePath + "/private/feed/unpacked/yara/{yara_rule}"
	localVarPath = strings.Replace(localVarPath, "{"+"yara_rule"+"}", _neturl.PathEscape(parameterToString(r.yaraRule, "")), -1)

	localVarHeaderParams := make(map[string]string)
	localVarQueryParams := _neturl.Values{}
	localVarFormParams := _neturl.Values{}

	// to determine the Content-Type header
	localVarHTTPContentTypes := []string{}

	// set Content-Type header
	localVarHTTPContentType := selectHeaderContentType(localVarHTTPContentTypes)
	if localVarHTTPContentType != "" {
		localVarHeaderParams["Content-Type"] = localVarHTTPContentType
	}

	// to determine the Accept header
	localVarHTTPHeaderAccepts := []string{"application/json"}

	// set Accept header
	localVarHTTPHeaderAccept := selectHeaderAccept(localVarHTTPHeaderAccepts)
	if localVarHTTPHeaderAccept != "" {
		localVarHeaderParams["Accept"] = localVarHTTPHeaderAccept
	}
	if r.ctx != nil {
		// API Key Authentication
		if auth, ok := r.ctx.Value(ContextAPIKeys).(map[string]APIKey); ok {
			if apiKey, ok := auth["api_key"]; ok {
				var key string
				if apiKey.Prefix != "" {
					key = apiKey.Prefix + " " + apiKey.Key
				} else {
					key = apiKey.Key
				}
				localVarHeaderParams["Authorization"] = key
			}
		}
	}
	req, err := a.client.prepareRequest(r.ctx, localVarPath, localVarHTTPMethod, localVarPostBody, localVarHeaderParams, localVarQueryParams, localVarFormParams, localVarFormFileName, localVarFileName, localVarFileBytes)
	if err != nil {
		return localVarReturnValue, nil, err
	}

	localVarHTTPResponse, err := a.client.callAPI(req)
	if err != nil || localVarHTTPResponse == nil {
		return localVarReturnValue, localVarHTTPResponse, err
	}

	localVarBody, err := _ioutil.ReadAll(localVarHTTPResponse.Body)
	localVarHTTPResponse.Body.Close()
	if err != nil {
		return localVarReturnValue, localVarHTTPResponse, err
	}

	if localVarHTTPResponse.StatusCode >= 300 {
		newErr := GenericOpenAPIError{
			body:  localVarBody,
			error: localVarHTTPResponse.Status,
		}
		return localVarReturnValue, localVarHTTPResponse, newErr
	}

	err = a.client.decode(&localVarReturnValue, localVarBody, localVarHTTPResponse.Header.Get("Content-Type"))
	if err != nil {
		newErr := GenericOpenAPIError{
			body:  localVarBody,
			error: err.Error(),
		}
		return localVarReturnValue, localVarHTTPResponse, newErr
	}

	return localVarReturnValue, localVarHTTPResponse, nil
}

type ApiGetPrivateFeedYaraTagsRequest struct {
	ctx _context.Context
	ApiService *FeedApiService
}


func (r ApiGetPrivateFeedYaraTagsRequest) Execute() (PrivateFeedYaraTags, *_nethttp.Response, error) {
	return r.ApiService.GetPrivateFeedYaraTagsExecute(r)
}

/*
 * GetPrivateFeedYaraTags Get list of yara tags in feed
 * Returns list of all yara tags available for the feed
 * @param ctx _context.Context - for authentication, logging, cancellation, deadlines, tracing, etc. Passed from http.Request or context.Background().
 * @return ApiGetPrivateFeedYaraTagsRequest
 */
func (a *FeedApiService) GetPrivateFeedYaraTags(ctx _context.Context) ApiGetPrivateFeedYaraTagsRequest {
	return ApiGetPrivateFeedYaraTagsRequest{
		ApiService: a,
		ctx: ctx,
	}
}

/*
 * Execute executes the request
 * @return PrivateFeedYaraTags
 */
func (a *FeedApiService) GetPrivateFeedYaraTagsExecute(r ApiGetPrivateFeedYaraTagsRequest) (PrivateFeedYaraTags, *_nethttp.Response, error) {
	var (
		localVarHTTPMethod   = _nethttp.MethodGet
		localVarPostBody     interface{}
		localVarFormFileName string
		localVarFileName     string
		localVarFileBytes    []byte
		localVarReturnValue  PrivateFeedYaraTags
	)

	localBasePath, err := a.client.cfg.ServerURLWithContext(r.ctx, "FeedApiService.GetPrivateFeedYaraTags")
	if err != nil {
		return localVarReturnValue, nil, GenericOpenAPIError{error: err.Error()}
	}

	localVarPath := localBasePath + "/private/feed/unpacked/yara"

	localVarHeaderParams := make(map[string]string)
	localVarQueryParams := _neturl.Values{}
	localVarFormParams := _neturl.Values{}

	// to determine the Content-Type header
	localVarHTTPContentTypes := []string{}

	// set Content-Type header
	localVarHTTPContentType := selectHeaderContentType(localVarHTTPContentTypes)
	if localVarHTTPContentType != "" {
		localVarHeaderParams["Content-Type"] = localVarHTTPContentType
	}

	// to determine the Accept header
	localVarHTTPHeaderAccepts := []string{"application/json"}

	// set Accept header
	localVarHTTPHeaderAccept := selectHeaderAccept(localVarHTTPHeaderAccepts)
	if localVarHTTPHeaderAccept != "" {
		localVarHeaderParams["Accept"] = localVarHTTPHeaderAccept
	}
	if r.ctx != nil {
		// API Key Authentication
		if auth, ok := r.ctx.Value(ContextAPIKeys).(map[string]APIKey); ok {
			if apiKey, ok := auth["api_key"]; ok {
				var key string
				if apiKey.Prefix != "" {
					key = apiKey.Prefix + " " + apiKey.Key
				} else {
					key = apiKey.Key
				}
				localVarHeaderParams["Authorization"] = key
			}
		}
	}
	req, err := a.client.prepareRequest(r.ctx, localVarPath, localVarHTTPMethod, localVarPostBody, localVarHeaderParams, localVarQueryParams, localVarFormParams, localVarFormFileName, localVarFileName, localVarFileBytes)
	if err != nil {
		return localVarReturnValue, nil, err
	}

	localVarHTTPResponse, err := a.client.callAPI(req)
	if err != nil || localVarHTTPResponse == nil {
		return localVarReturnValue, localVarHTTPResponse, err
	}

	localVarBody, err := _ioutil.ReadAll(localVarHTTPResponse.Body)
	localVarHTTPResponse.Body.Close()
	if err != nil {
		return localVarReturnValue, localVarHTTPResponse, err
	}

	if localVarHTTPResponse.StatusCode >= 300 {
		newErr := GenericOpenAPIError{
			body:  localVarBody,
			error: localVarHTTPResponse.Status,
		}
		return localVarReturnValue, localVarHTTPResponse, newErr
	}

	err = a.client.decode(&localVarReturnValue, localVarBody, localVarHTTPResponse.Header.Get("Content-Type"))
	if err != nil {
		newErr := GenericOpenAPIError{
			body:  localVarBody,
			error: err.Error(),
		}
		return localVarReturnValue, localVarHTTPResponse, newErr
	}

	return localVarReturnValue, localVarHTTPResponse, nil
}
